apps/services/user-profile/src/app/v2/islykill.service.ts

```diff
@@ -1,11 +1,49 @@
-import { Injectable, InternalServerErrorException } from '@nestjs/common'
-
+import {
+  BadRequestException,
+  Injectable,
+  InternalServerErrorException,
+} from '@nestjs/common'
 import { IslyklarApi, PublicUser } from '@island.is/clients/islykill'
 
 @Injectable()
 export class IslykillService {
   constructor(private readonly islyklarApi: IslyklarApi) {}
 
+  async getIslykillSettings(
+    nationalId: string,
+  ): Promise<PublicUser & { noUserFound?: boolean }> {
+    try {
+      const userData: PublicUser = await this.islyklarApi.islyklarGet({
+        ssn: nationalId,
+      })
+
+      return {
+        ssn: nationalId,
+        email: userData.email,
+        mobile: userData.mobile,
+        bankInfo: userData.bankInfo,
+        lastLogin: userData.lastLogin,
+        nextLastLogin: userData.nextLastLogin,
+        lastPassChange: userData.lastPassChange,
+        canNudge: userData.canNudge,
+        onlyCert: userData.onlyCert,
+        nudgeLastAsked: userData.nudgeLastAsked,
+      }
+    } catch ({ status, message }) {
```

Comment:
Is this correctly typed? Isn't this unknown?

You could do 
```
if (err instanceof Error) {
    // Now TS correctly types the error
  }
```

Reply:
You will probably also have to type the status field

Reply:
Can I not just do `err: Error` should that not result in the same instance ?